vars/gh.groovy

```diff
@@ -26,35 +26,58 @@
 
 */
 
+import groovy.transform.Field
+
+@Field def ghLocation = ''
+
 def call(Map args = [:]) {
   if(!isUnix()) {
     error 'gh: windows is not supported yet.'
   }
   def command = args.containsKey('command') ? args.command : error('gh: command argument is required.')
   def credentialsId = args.get('credentialsId', '2a9602aa-ab9f-4e52-baf3-b71ca88469c7')
-  def flags = args.get('flags', [])
-  withCredentials([string(credentialsId: "${credentialsId}", variable: 'GITHUB_TOKEN')]) {
-    def flagsCommand = ''
-    if (flags) {
-      flags.each { k, v ->
-        if (v instanceof List) {
-          v.each { value ->
-            flagsCommand += "--${k}=${value} "
-          }
-        } else {
-          flagsCommand += "--${k}=${v} "
-        }
-      }
+  def flags = args.get('flags', [:])
+
+  // Use the current location as the git repo otherwise uses the env variables to pass
+  // the repo information to the gh command
+  def isGitWorkspace = sh(label: 'isGitWorkspace', script: 'git rev-list HEAD -1 1> /dev/null 2>&1', returnStatus: true) == 0
+  if (isGitWorkspace) {
+    log(level: 'DEBUG', text: 'gh: running within a git workspace.')
+  } else {
+    log(level: 'DEBUG', text: 'gh: running outside of a git workspace. Using REPO_NAME and ORG_NAME if they are set')
+    if (env.REPO_NAME?.trim() && env.ORG_NAME?.trim()) {
+      flags['repo'] = "${env.ORG_NAME}/${env.REPO_NAME}"
```

Comment:
Whether to use the existing git details from the current directory otherwise fall back to the env variables.